lib/core/test/unit/Cardano/Wallet/ApiSpec.hs

```diff
@@ -218,31 +218,37 @@ instance (GenericApiSpec a, GenericApiSpec b) => GenericApiSpec (a :<|> b) where
 instance GenericApiSpec Request where
     gSpec _ _ = pure ()
 
+instance GenericApiSpec a => GenericApiSpec [a] where
+    gSpec xs toSpec = foldr (\x y -> gSpec x toSpec >> y) (pure ()) xs
+
 instance
     ( Typeable a, Malformed (PathParam a)
-    ) => GenericApiSpec (PathParam a -> Request)
+    ) => GenericApiSpec (PathParam a -> [Request])
   where
-    gSpec toRequest toSpec = toSpec $
-        SomeTest (Proxy @a) (first toRequest <$> malformed @(PathParam a))
+    gSpec toRequest toSpec = do
+        let tests = fmap (\(xs, e) -> fmap (,e) xs)
+                (first toRequest <$> malformed @(PathParam a))
```

Comment:
Perhaps it's just me, but I do have a bit of trouble imagining what is happening here.

I wonder if the code could be made simpler with a nested `forM_`?

Reply:
This code is type tetris... I doubt changing the functions will make it any more accessible. We can add more type signatures though.